### PR TITLE
fix: schedule unscheduled emails with tasker in cron job

### PR DESCRIPTION
**Pull request by CarinaWolli**
## Summary by mrge
Updated the cron job to handle unscheduled email reminders with the tasker service instead of SendGrid, ensuring smooth transition when SendGrid environment variables are removed. 

**Migration**
- Removes SendGrid dependency checks and adds fallback to schedule all unscheduled reminders with the tasker
- Only needs to run once to migrate existing unscheduled reminders

- Fixes CAL-5516

### How should this be tested?
- Ensure there are unscheduled Reminders in the DB
- Remove SendGrid .env vars
- Trigger the cron job endpoint: POST /api/cron/workflows/scheduleEmailReminders
- All unscheduled Reminders should now be scheduled with the Tasker

### CODE REVIEW

File: packages/features/ee/workflows/api/scheduleEmailReminders.ts [hunk 1]
```diff
@@ -398,19 +431,32 @@ export async function handler(req: NextRequest) {
...
-              sendAt: dayjs(reminder.scheduledDate).unix(),
-              replyTo: reminder.booking?.userPrimaryEmail ?? reminder.booking.user?.email,
-              sender: reminder.workflowStep?.sender,
-            })
-          );
+          const batchId = isSendgridEnabled ? await getBatchId() : undefined;
+
+          const mailData = {
+            subject: emailContent.emailSubject,
+            to: [sendTo],
```
[the remark this answers is not in the document]
Comment (CarinaWolli) [reply packages/features/ee/workflows/api/scheduleEmailReminders.ts:438]: `sendTo` is a string here

File: packages/features/ee/workflows/api/scheduleEmailReminders.ts [hunk 2]
```diff
@@ -21,7 +21,9 @@
   getAllRemindersToCancel,
   getAllRemindersToDelete,
   getAllUnscheduledReminders,
+  select,
 } from "../lib/getWorkflowReminders";
+import { sendOrScheduleWorkflowEmails } from "../lib/reminders/providers/emailProvider";
```
[the remark this answers is not in the document]
Comment (CarinaWolli) [reply packages/features/ee/workflows/api/scheduleEmailReminders.ts]: fixed

File: packages/features/tasker/repository.ts [hunk 3]
```diff
@@ -154,6 +154,23 @@ export class Task {
     });
   }
 
+  static async cancelWithReference(referenceUid: string, type: TaskTypes) {
+    const task = await db.task.findFirst({
```
[the remark this answers is not in the document]
Comment (CarinaWolli) [reply packages/features/tasker/repository.ts:158]: fixed

**Review by zomars — APPROVED** [review]

**Review by zomars — CHANGES_REQUESTED** [review]
I noticed a few potential issues that might be causing test failures:

1. The function `sendOrScheduleWorkflowEmails` was renamed to `sendOrScheduleWorkflowEmail` (singular), but all references need to be updated consistently across the codebase and tests.

2. The parameter handling between SendGrid and tasker implementations differs slightly - particularly around how `referenceUid` is managed. Make sure all code paths properly handle both cases.

3. The `deleteScheduledEmailReminder` function now has conditional logic for tasker vs SendGrid, but there might be edge cases not covered in tests.

4. The type definitions and interfaces might need updates to reflect the new function signatures and parameters.

These changes touch core scheduling functionality, so tests that rely on the previous behavior might need updates.

**Comment by CarinaWolli** [comment]
@zomars everything is passing now

**Review by zomars — APPROVED** [review]
